.github/workflows/ensnode_railway_deploy.yml

```
@@ -0,0 +1,111 @@
+                echo "SVC_ID="${{ secrets.SUBGRAPH_SVC_ID }} >> "$GITHUB_ENV"
+                ;;
+              *)
+                echo "Branch not recognized, skipping workflow"
+                exit 1
+            esac
+
+      - name: Build & Push Docker image for ENSNode to Github Container registry
+        env:
+          IMAGE_NAME: ghcr.io/${{ github.repository }}/ensnode
+        run: |
+          echo ${{ secrets.GITHUB_TOKEN }} | docker login ghcr.io -u ${{ github.actor }} --password-stdin
+          docker build -f apps/ensnode/Dockerfile -t $IMAGE_NAME:$TAG .
+          docker push $IMAGE_NAME:$TAG
+
+      - name: Determine Railway variable to be updated
+        run: |
+          TAG=${{ env.TAG }}
+          VARIABLE_NAME=${TAG^^}_DATABASE_SCHEMA
+          echo "Looking for: ${VARIABLE_NAME} environment variable"
+          echo "VARIABLE_NAME=$VARIABLE_NAME" >> "$GITHUB_ENV"
+      
+      - name: Get current Railway schema
+        run: |
+          # Send Railway API request to get all shared variables. We need those to check if we really need to redeploy.
+          API_RESPONSE=`curl --request POST \
+          --fail \
+          --silent \
+          --url https://backboard.railway.app/graphql/v2 \
+          --header 'Project-Access-Token: '${{ env.RAILWAY_TOKEN }} \
+          --header 'Content-Type: application/json' \
+          --data '{
+            "query": "query variables($projectId: String!, $environmentId: String!) { variables(projectId: $projectId, environmentId: $environmentId) }",
+              "variables": {
+                "projectId": "'${{ env.RAILWAY_PROJECT_ID }}'" ,
+                "environmentId": "'${{ env.RAILWAY_ENVIRONMENT_ID }}'"
+              }
+          }'`
+          # Filter all env variables from response except one that is defining schema for specific deployment.
+          RAILWAY_SCHEMA=`echo ${API_RESPONSE} | jq '.data.variables.'${{ env.VARIABLE_NAME }}`
+          echo "RAILWAY_SCHEMA=$RAILWAY_SCHEMA" >> "$GITHUB_ENV"
+          
+
+      - name: Update Railway schema variable
+        # Check if newly calculated schema matches old one, if not replace env variable and force deployment. Otherwise skip deployment.
+        run: |
+          # Sending Railway API request to update specifc shared variable
+          if [[ ${{env.RAILWAY_SCHEMA}} != ${{ env.DATABASE_SCHEMA }} ]]; then
+          curl --request POST \
+              --silent \
+              --fail \
+              --url https://backboard.railway.app/graphql/v2 \
+              --header 'Project-Access-Token: '${{ env.RAILWAY_TOKEN }} \
+              --header 'Content-Type: application/json' \
+              --data '{"query": "mutation variableUpsert { variableUpsert(input: { projectId: \"'${{ env.RAILWAY_PROJECT_ID }}'\", environmentId: \"'${{ env.RAILWAY_ENVIRONMENT_ID }}'\", name: \"'${VARIABLE_NAME}'\", value: \"${{ env.DATABASE_SCHEMA }}\" }) }"}'
+          # Sending Railway API request to redeploy service
```

```suggestion
          # Sending Railway API request to redeploy service (using the newly updated configuration)
```

Is that fair? In other words, the first `curl` seems to change something, but not necessarily to trigger a redeployment? And therefore we need this second `curl` to trigger the redeployment using the new config?